internal/base_backup.go

```diff
@@ -0,0 +1,255 @@
+package internal
+
+/*
+This object represents a base backup object.
+A base backup object can connect to Postgres, issue a BASE_BACKUP command, and receive the backup data from Postgres.
+*/
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pglogrepl"
+	"github.com/jackc/pgproto3/v2"
+	"github.com/pkg/errors"
+	"github.com/wal-g/storages/storage"
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/internal/ioextensions"
+)
+
+var (
+	tbsOutOfRange     error = errors.New("requesting next tablespace after all tablespaces are streamed")
+	tbsStillStreaming error = errors.New("Cannot move to next tablse. Current tablespace is not yet fully streamed.")
+)
+
+// The BaseBackup object represents a Postgres streaming BASE_BACKUP, connecting to Postgres, and streaming backup data.
```

Comment:
Maybe add some prefix like StreamingBaseBackup or RemoteBaseBackup to avoid confusion with the regular WAL-G base backup?

Reply:
Done